components/centraldashboard/app/api.ts

```diff
@@ -181,33 +268,170 @@ export class Api {
+                  err,
+                });
+              }
+            })
         .post('/create-workgroup', async (req: Request, res: Response) => {
-          if (!req.user.hasAuth) {
-            res.status(405).json(OPERATION_NOT_SUPPORTED);
-            return;
-          }
-
-          const profile = req.body as CreateProfileRequest;
-          try {
-            // Use the request body if provided, fallback to auth headers
-            await this.profilesService.createProfile({
-              metadata: {
-                name: profile.namespace || req.user.username,
-              },
-              spec: {
-                owner: {
-                  kind: 'User',
-                  name: profile.user || req.user.email,
+              if (!req.user.hasAuth) {
+                return apiError({
+                  res,
+                  code: 405,
+                  error: ERRORS.operation_not_supported,
+                });
+              }
+    
+              const profile = req.body as CreateProfileRequest;
+              try {
+                const namespace = profile.namespace || req.user.username;
+                // Use the request body if provided, fallback to auth headers
+                await this.profilesService.createProfile({
+                  metadata: {
+                    name: namespace,
+                  },
+                  spec: {
+                    owner: {
+                      kind: 'User',
+                      name: profile.user || req.user.email,
+                    }
+                  },
+                });
+                res.json({message: `Created namespace ${namespace}`});
+              } catch (err) {
+                this.surfaceProfileControllerErrors({
+                  res,
+                  msg: 'Unexpected error creating profile',
+                  err,
+                });
+              }
+            })
+        .post(
+              '/add-contributor/:namespace',
+              async (req: Request, res: Response) => {
+                const {namespace} = req.params;
+                const {contributor} = req.body as AddOrRemoveContributorRequest;
+                if (!contributor || !namespace) {
+                  return apiError({
+                    res,
+                    error: `Missing contributor / namespace fields.`,
+                  });
+                }
+                if (!EMAIL_RGX.test(contributor)) {
+                  return apiError({
+                    res,
+                    error: `Contributor doesn't look like a valid email address`,
+                  });
+                }
+                let errIndex = 0;
+                try {
+                  const binding = this.mapSimpleBindingToWorkgroupBinding({
+                    user: contributor,
+                    namespace,
+                    role: 'contributor',
+                  });
+                  const {headers} = req;
+                  delete headers['content-length'];
```

Comment:
I see what you meant about the content-length if you were taking the full headers from the incoming request. I assume you don't need anything other than the authentication headers. Couldn't you just take the `req.user.auth` property which extracted the original headers?

Reply:
Sure, I'll go ahead and do that.

Reply:
Looking at the kfam code, it definitely seems like they only use the [email header](https://github.com/kubeflow/kubeflow/blob/f3b8d2337e8b29ac213fbf94e29a3783c76b5a90/components/access-management/kfam/api_default.go#L227). I'd be reluctant to pass other headers in unnecessarily. Obviously I think you saw already the difficulty in debugging an issue with just `Content-Length`. I'd hate to see a scenario like that appear in a production environment due to some unforeseen interaction with the Kfam server's treatment of the headers.